src/comp/comp.py

```diff
@@ -24,48 +30,55 @@ def __repr__(self):
 # Write a list comprehension that creates a list of names of everyone
 # whose name starts with 'D':
 print("Starts with D:")
-a = []
+# or can use human.name.startswith("D")
+a = [human.name for human in humans if human.name[0].lower() == "d"]
+
 print(a)
 
 # Write a list comprehension that creates a list of names of everyone
 # whose name ends in "e".
 print("Ends with e:")
-b = []
+b = [human.name for human in humans if human.name.endswith("e")]
 print(b)
 
 # Write a list comprehension that creates a list of names of everyone
 # whose name starts with any letter between 'C' and 'G' inclusive.
+# needs to be a tuple (), not a list [] to use in startswith() method
+range = ("C", "D", "E", "F", "G")
 print("Starts between C and G, inclusive:")
-c = []
+c = [human.name for human in humans if human.name.startswith(range)]
 print(c)
 
 # Write a list comprehension that creates a list of all the ages plus 10.
 print("Ages plus 10:")
-d = []
+d = [(human.age + 10) for human in humans]
 print(d)
 
 # Write a list comprehension that creates a list of strings which are the name
 # joined to the age with a hyphen, for example "David-31", for all humans.
 print("Name hyphen age:")
-e = []
+e = [str(f'{human.name}-{human.age}') for human in humans]
 print(e)
 
 # Write a list comprehension that creates a list of tuples containing name and
 # age, for example ("David", 31), for everyone between the ages of 27 and 32,
 # inclusive.
 print("Names and ages between 27 and 32:")
-f = []
+f = [(human.name, human.age)
+     for human in humans if human.age > 26 and human.age < 33]
 print(f)
 
 # Write a list comprehension that creates a list of new Humans like the old
 # list, except with all the names uppercase and the ages with 5 added to them.
 # The "humans" list should be unmodified.
 print("All names uppercase:")
-g = []
+#g = [str(f'{human.name.upper()}, {human.age + 5}') for human in humans]
+g = [Human(human.name.upper(), human.age + 5) for human in humans]
 print(g)
+# print(humans)
 
 # Write a list comprehension that contains the square root of all the ages.
```

Comment:
You did an excellent job in your list comprehension exercises. Everything is easy to read